com.ibm.wala.core/src/com/ibm/wala/ipa/slicer/json/SDGSerializer.java

```diff
@@ -0,0 +1,98 @@
+package com.ibm.wala.ipa.slicer.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.ibm.wala.ipa.callgraph.CGNode;
+import com.ibm.wala.ipa.callgraph.CallGraph;
+import com.ibm.wala.ipa.callgraph.propagation.InstanceKey;
+import com.ibm.wala.ipa.slicer.PDG;
+import com.ibm.wala.ipa.slicer.SDG;
+import com.ibm.wala.ipa.slicer.Statement;
+import com.ibm.wala.util.collections.Iterator2Collection;
+import com.ibm.wala.util.graph.traverse.Topological;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * Serializes an SDG to JSON. All edges are explicitly materialized and stored.
+ * For edge serialization, nodes are encoded as [ i, j ] where i is the id of
+ * the PDG the node belongs to and j is its local id within the PDG.
+ *
+ */
+public class SDGSerializer extends StdSerializer<SDG<? extends InstanceKey>> {
```

Comment:
Would it make sense to be able to build a more condensed JSON file? E.g., a JSON file with no spaces, no new lines, and very compact field names (for instance : predecessors -> p, successors -> s, etc.). I believe we could get a significant reduction in size of the final JSON.